### PR TITLE
noti: init at 3.1.0

### PR DESCRIPTION
**Pull request by stites**
###### Motivation for this change
Add a new, works-out-of-the-box notifier

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions: Ubunt17.04
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

First-time submitter! let me know if I should move this package somewhere else, or if anything is wrong.

### CODE REVIEW

**Review by srhb — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/noti/default.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{ stdenv, fetchzip }:
+
+stdenv.mkDerivation rec {
+  name = "noti-${version}";
+  version = "3.1.0";
+
+  src = fetchzip {
+    url = "https://github.com/variadico/noti/releases/download/${version}/noti${version}.linux-amd64.tar.gz";
```
[the remark this answers is not in the document]
Comment (stites) [reply pkgs/tools/misc/noti/default.nix]: Nope, I just figured that prebuilt binaries were preferable to the source code -- I'll fix that now!

**Comment by stites** [comment]
@srhb -- Please take a look! I'm a bit iffy on the proper way to set up documentation -- I think the `postInstall` is correct (as in "works on my machine"), but I'm not sure if there's a better way to do this.

I'm also a bit iffy on go in general, but luckily nix makes it easy to figure out the build process. : )

File: pkgs/tools/misc/noti/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  src = fetchFromGitHub {
+    owner = "variadico";
+    repo = "noti";
+    rev = "${version}";
+    sha256 = "1chsqfqk0pnhx5k2nr4c16cpb8m6zv69l1jvv4v4903zgfzcm823";
+  };
+
+  goPackagePath = "github.com/variadico/noti";
+
+  goDeps = ./deps.nix;
```
Comment (srhb) [inline pkgs/tools/misc/noti/default.nix]: The deps appear to be vendored in the tarball fetched by `fetchFromGitHub`. Can you check if `goDeps` are really necessary?

**Review by srhb — CHANGES_REQUESTED** [review]

**Comment by stites** [comment]
noti, especially, has a decently sized Makefile which originally lead me to believe it had a non-standard build. This repo doesn't seem to output the man pages, anyhow, and they come pre-build in the repo.

**Comment by srhb** [comment]
Thanks@stites. One last thing: Please squash this into one commit following the rules at https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes (ie. "noti: init at 3.1.0") or two commits, one for this addition and one "add stites to maintainer list" :)

Thanks a lot for your effort, and welcome to the nixpkgs maintainers list. :smiley:

**Comment by stites** [comment]
Thank you for the guidance, this was a lot of fun!

**Pull request merged**